modules/01 JavaScript Programming/Assignments/19/answer19.js

```diff
@@ -0,0 +1,87 @@
+
+function validateCreditCard (creditCard)
+{
+ //   var creditCard=9999777788880000;
+    let arrayCreditCard = Array.from(creditCard.toString());
+    let  answer={
+        valid: false,
+        number: creditCard,
+        error: ""
+    }
+        
+        // remove "-" from CreditCard number
+    if(arrayCreditCard[4]==='-' && arrayCreditCard[9] === '-'  &&arrayCreditCard[14]==='-'   )
+    {
+        arrayCreditCard.splice(4, 1);
+        arrayCreditCard.splice(8, 1);
+        arrayCreditCard.splice(12, 1);
+
+    }
+   // console.log(arrayCreditCard);
+  //  var strCreditCard=creditCard.toString();
+    if(arrayCreditCard.length!=16)
+    {
+        answer["valid"]= false;
+        answer["error"]= "wrong length";
+        return answer;
+    }
+    if(arrayCreditCard[arrayCreditCard.length-1]%2!=0)
+    {
+        answer["valid"]= false;
+        answer["error"]= "odd final number";
+        return answer;
+    }
+   
+    if(arrayCreditCard.every( function (digit)
+    {
+  
+        return (digit.valueOf() >=0 && digit.valueOf()<10);
+    })===false)
+    {
+        answer["valid"]= false;
+        answer["error"]= "invalid characters";
+        return answer;
+    }
+   
+    if(arrayCreditCard.every( function (digit)
+    {
+
+        return arrayCreditCard[0]===digit;
+    }))
+    {
+        
+        answer["valid"]= false;
+        answer["error"]= "invalid characters";
+        return answer;
+    }
+    var sum=0;
+    for(var i=0;i<arrayCreditCard.length;++i)
+    {
+        //second + is to convert string to a number 
+        sum+=+arrayCreditCard[i].valueOf();
+    }
+    if(sum>16)
+    {
+        answer["valid"]= true;
+        return answer;
+    }
+    else{
+        answer["valid"]= false;
+        answer["error"]= "sum less than 16";
+        return answer;
+
+    }
+  
+
+
+}
+console.log(validateCreditCard("9999777788880000"));
```

Comment:
A valid output shouldn't include an "error" key:
{valid: true, number: "9999777788880000", error: ""}

The example says:
{ valid: true, number: 'a923-3211-9c01-1112' }